docs/src/codegen-intro.md

```diff
@@ -28,18 +35,82 @@ playwright codegen demo.playwright.dev/todomvc
+
+Use the **clear** button to clear the code to start recording again. Once finished close the Playwright inspector window or stop the terminal command.
+
+To learn more about generating tests check out or detailed guide on [Codegen](./codegen.md).
+### Recording a test
+* langs: java
+
+Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at your page and figure out the best locator, prioritizing role, text and test id locators. If the generator finds multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, so you don't have to worry about failing tests due to locators.
+
+<img width="1365" alt="Recording a test" src="https://user-images.githubusercontent.com/13063165/212754804-0d9f9d52-0a48-45c8-970d-e672d4a91221.png" />
 
 When you have finished interacting with the page, press the **record** button to stop the recording and use the **copy** button to copy the generated code to your editor.
 
 Use the **clear** button to clear the code to start recording again. Once finished close the Playwright inspector window or stop the terminal command.
 
 To learn more about generating tests check out or detailed guide on [Codegen](./codegen.md).
+### Recording a test
+* langs: python
+
+Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at your page and figure out the best locator, prioritizing role, text and test id locators. If the generator finds multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, so you don't have to worry about failing tests due to locators.
+
+<img width="1365" alt="Recording a test" src="https://user-images.githubusercontent.com/13063165/212751993-b7da2c40-a7cc-4b13-9a91-40ee837042a1.png" />
+
+When you have finished interacting with the page, press the **record** button to stop the recording and use the **copy** button to copy the generated code to your editor.
+
+Use the **clear** button to clear the code to start recording again. Once finished close the Playwright inspector window or stop the terminal command.
+
+To learn more about generating tests check out or detailed guide on [Codegen](./codegen.md).
+### Recording a test
+* langs: csharp
+
+Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at your page and figure out the best locator, prioritizing role, text and test id locators. If the generator finds multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, so you don't have to worry about failing tests due to locators.
```

Comment:
```suggestion
Run `codegen` and perform actions in the browser. Playwright will generate the code for the user interactions. `Codegen` will look at the rendered page and figure out the recommended locator, prioritizing role, text and test id locators. If the generator identifies multiple elements matching the locator, it will improve the locator to make it resilient and uniquely identify the target element, therefore eliminating and reducing test/s failing and flaking due to locators
```